prac_08/squaring.py

```diff
@@ -0,0 +1,33 @@
+"""
+CP1404/CP5632 Practical
+Kivy GUI program to square a number
+Lindsay Ward, IT@JCU
+Started 13/10/2015
+"""
+
+from kivy.app import App
+from kivy.lang import Builder
+from kivy.core.window import Window
+
+__author__ = 'Lindsay Ward'
+
+
+class SquareNumberApp(App):
+    """ SquareNumberApp is a Kivy App for squaring a number """
+    def build(self):
+        """ build the Kivy app from the kv file """
+        Window.size = (200, 100)
+        self.title = "Square Number"
+        self.root = Builder.load_file('squaring.kv')
+        return self.root
+
+    def handle_calculate(self, value):
+        """ handle calculation (could be button press or other call), output result to label widget """
+        try:
+            result = float(value) ** 2
+            self.root.ids.output_label.text = str(result)
+        except ValueError:
+            pass
```

Comment:
Looks great here too
Alternatively for ValueError, you can also consider self.root.ids.output_label.text = "" so that it counts as a blank instead of completely passing, which doesn't change anything just for this exercise honestly, but just another way of doing it in the larger scale of things in case you need to do more with ValueError than just passing (e.g. taking a default value instead of stopping program).

Reply:
Yes, but this part was supposedly the "improved" and better version, as stated by the instructions in the practical so I just left it as it is. Thanks, though!